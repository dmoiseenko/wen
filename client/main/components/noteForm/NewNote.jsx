import React from 'react';
import PropTypes from 'prop-types';


export default function NewNote({ addNote }) {
  function handleClick(e) {
    e.preventDefault();
    addNote();
  }

  return (
    <p className="control">
      <button
        type="submit"
        className="button is-medium is-primary"
        onClick={handleClick}
      >
        Add Note
      </button>
    </p>
  );
}

NewNote.propTypes = {
  addNote: PropTypes.func.isRequired,
};
