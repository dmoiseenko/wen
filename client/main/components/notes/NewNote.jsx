import React from 'react';
import PropTypes from 'prop-types';


export default function NewNote({ addNote, noteText }) {
  function handleClick(e) {
    e.preventDefault();
    addNote(noteText);
  }

  return (
    <p className="control">
      <button
        type="submit"
        className="button is-primary"
        onClick={handleClick}
      >
        Add Note
      </button>
    </p>
  );
}

NewNote.propTypes = {
  addNote: PropTypes.func.isRequired,
  noteText: PropTypes.string.isRequired
};
