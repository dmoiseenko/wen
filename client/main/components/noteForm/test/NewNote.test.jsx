import React from 'react';
import { shallow } from 'enzyme';

import NewNote from '../NewNote';


it('should render', () => {
  const props = {
    addNote: jest.fn()
  };

  const wrapper = shallow(
    <NewNote {...props} />
  );

  expect(wrapper.node).toMatchSnapshot();
});

it('should call addNote when click fires', () => {
  const addNote = jest.fn();
  const event = {
    preventDefault: jest.fn()
  };

  const wrapper = shallow(
    <NewNote
      addNote={addNote}
    />,
  );

  wrapper.find('.button').simulate('click', event);

  expect(event.preventDefault).toHaveBeenCalled();
  expect(addNote).toHaveBeenCalled();
});
