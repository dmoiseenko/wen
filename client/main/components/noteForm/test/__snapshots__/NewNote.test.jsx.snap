// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
<p
  className="control"
>
  <button
    className="button is-medium is-primary"
    onClick={[Function]}
    type="submit"
  >
    Add Note
  </button>
</p>
`;
